tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/validation/ContactDetailsValidator.java

```diff
@@ -124,14 +135,42 @@ private void validatePhoneNumber(String fieldName, String value, List<FieldError
 
   private List<FieldError> checkEmail(ContactDetailsDTO dto) {
     List<FieldError> fieldErrors = new ArrayList<>();
-    validateEmail("email", dto.getEmail(), fieldErrors);
+    String fieldName = "email";
+    validateEmailFormat(fieldName, dto.getEmail(), fieldErrors);
+    validateEmailUniqueness(fieldName, dto, fieldErrors);
     return fieldErrors;
   }
 
-  private void validateEmail(String fieldName, String value, List<FieldError> fieldErrors) {
+  private void validateEmailFormat(String fieldName, String value, List<FieldError> fieldErrors) {
     if (value != null && !value.matches(REGEX_EMAIL)) {
```

Comment:
Given that there's a TODO around improving this  on the REGEX_EMAIL string definition, I wonder if would be worth using the @Email validation, instead of a custom regex? There's a bit of a clunky implementation of this sort of idea here: https://github.com/Health-Education-England/TIS-GENERIC-UPLOAD/blob/19b4b0db688b0080b3233c1e208e3ded24119c18/generic-upload-service/src/main/java/com/transformuk/hee/tis/genericupload/service/service/PersonTransformerService.java#L899 . That would at least mean that the same rules are used in generic-upload and TCS.

I don't think its needed for this ticket, but it might be worth a thought.

Reply:
The @Email annotation is used in [ContactDetailsDTO](https://github.com/Health-Education-England/TIS-TCS/blob/1f4de2d0ec47bb58125b715b1c2f951ca82989c7/tcs-api/src/main/java/com/transformuk/hee/tis/tcs/api/dto/ContactDetailsDTO.java#L76). But in hibernate-validator-6.x, this annotation is deprecated (https://docs.jboss.org/hibernate/stable/validator/api/org/hibernate/validator/constraints/Email.html), while generic upload service uses hibernate-validator-5.1.3. 

I've no idea why we need the custom regex. It really worths a check.